pkg/engine/engine.go

```diff
@@ -49,6 +50,7 @@ type Engine struct {
 	runners map[string]api.Runner
 	envcfg  *config.EnvConfig
 	ctx     context.Context
+	stor    *task.TaskStorage
```

Comment:
I put this here, but maybe this doesn't need to be part of the engine. Maybe this is better situated as part of the under the Service struct.